okio-files/src/macosX64Main/kotlin/okio/cinterop.kt

```diff
@@ -57,6 +57,22 @@ internal fun Buffer.read(
   }
 }
 
+internal inline fun <S : Source, T> S.use(block: (S) -> T): T {
+  try {
+    return block(this)
+  } finally {
+    close()
```

Comment:
Any idea what happens if this throws when `block` also threw?

Reply:
It incorrectly clobbers the thrown exception. Lemme fix up to do the correct thing and throw the first exception...